src/nanoarrow/common/inline_types.h

```diff
@@ -808,6 +815,15 @@ struct ArrowArrayView {
   /// type_id == union_type_id_map[128 + child_index]. This value may be
   /// NULL in the case where child_id == type_id.
   int8_t* union_type_id_map;
+
+  /// \brief Number of variadic buffers
```

Comment:
@paleolimbot this PR is not quite ready for a full PR review, but figured I'd ping some more usability questions off you.

As you suggested, I took these members from the great work that @jorisvandenbossche started. Assuming we have `n_buffers` of 4, two of those buffers are accessible from the `buffer_views` member (the validity mask and inline/prefix buffer), one buffer is accessible from `variadic_buffer_views`, and the last is accessible from `variadic_buffer_sizes`

Is that what you had in mind?